catalyst-gateway-crates/c509-certificate/src/c509_extensions/alt_name.rs

```diff
@@ -0,0 +1,114 @@
+//! C509 Alternative Name uses for Subject Alternative Name extension and
+//! Issuer Alternative Name extension.
+
+use minicbor::{encode::Write, Decode, Decoder, Encode, Encoder};
+
+use crate::c509_general_names::GeneralNames;
+
+/// Enum for type that can be a `GeneralNames` or a text.
+#[derive(Debug, Clone, PartialEq)]
+pub enum GeneralNamesOrText {
+    GeneralNames(GeneralNames),
+    Text(String),
+}
+
+/// Alternative Name extension.
+/// Can be interpreted as a GeneralNames / text
+///
+/// # Fields
+/// * value - A value of alternative name that can be either a `GeneralNames` or a text.
+#[derive(Debug, Clone, PartialEq)]
+pub struct AltName {
+    value: GeneralNamesOrText,
+}
```

Comment:
You should use `AltName` as an enum instead of wrapping the value of `GeneralNamesOrText`. This will reduce numbers of structs / enum to maintain separately.